sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/jsonschema/JsonSchemaResourceIntegrationTest.java

```diff
@@ -0,0 +1,198 @@
+        .await()
+        .atMost(Duration.ofSeconds(10));
+
+    tableManager
+        .createCollectionTable(DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+        .await()
+        .atMost(Duration.ofSeconds(10));
+  }
+
+  @Nested
+  @Order(1)
+  @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+  class PutJsonSchema {
+
+    @Test
+    @Order(1)
+    public void happyPath() throws JsonProcessingException {
+      String body =
+          """
+                        {\"$schema\": \"https://json-schema.org/draft/2019-09/schema\"}
+                    """;
+
+      given()
+          .contentType(ContentType.JSON)
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .body(body)
+          .put(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .body("schema.$schema", is("https://json-schema.org/draft/2019-09/schema"));
+    }
+
+    @Test
+    @Order(2)
+    public void tableDoesNotExist() {
+      String body =
+          """
+            {\"$schema\": \"https://json-schema.org/draft/2019-09/schema\"}
+          """;
+
+      given()
+          .contentType(ContentType.JSON)
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .body(body)
+          .put(BASE_PATH, DEFAULT_NAMESPACE, "notatable")
+          .then()
+          .statusCode(400)
+          .body("code", is(400))
+          .body(
+              "description",
+              is(
+                  "Invalid argument for gRPC operation (INVALID_ARGUMENT->Bad Request): INVALID_ARGUMENT: unconfigured table notatable"));
+      ;
+    }
+
+    @Test
+    public void invalidJsonSchema() {
+      String body =
+          "{\n"
+              + "  \"$schema\": \"https://json-schema.org/draft/2019-09/schema\",\n"
+              + "  \"type\": \"object\",\n"
+              + "  \"properties\": {\n"
+              + "    \"something\": { \"type\": \"strin\" }\n"
+              + "  }\n"
+              + "}";
+
+      given()
+          .contentType(ContentType.JSON)
+          .body(body)
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .put(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(400)
+          .body("code", is(400))
+          .body(
+              "description",
+              is(
+                  "[error]: \"strin\" is not a valid primitive type (valid values are: [array, boolean, integer, null, number, object, string]); "));
+    }
+
+    @Test
+    public void noPayload() {
+      given()
+          .contentType(ContentType.JSON)
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .put(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(400)
+          .body("code", is(400))
+          .body("description", is("Request invalid: json schema not provided."));
+    }
+  }
+
+  @Nested
+  @Order(2)
+  class GetJsonSchema {
```

Comment:
missing cases:

* keyspace not existing
* not a valid document table (try with `system.something`)